File: ex1/python/computeCost.py
import numpy as np
def computeCost(X, y, theta):
#   Cost for linear regression
    h = X.dot( theta )
    a = (h - y).T
    b = h - y

    m = h.shape[0] # Number of trainning examples
    mean = 1 / (2 * m)

    J = mean * (a.dot( b ))
    return J

File: ex1/octave/computeCost.m
function J = computeCost(X, y, theta)
% Cost for linear regression
    h = X * theta;
    a = (h - y)';
    b = h - y ;

    m = length(y); % number of training examples
    mean = 1 / (2 * m);

    J = mean * a * b;

end
